pkg/ddevapp/router.go

```diff
@@ -87,17 +86,20 @@ func StartDdevRouter() error {
 	_, err = f.WriteString(doc.String())
 	util.CheckErr(err)
 
-	container, err := findDdevRouter()
-	// If we have a router running, we don't have to stop and start it.
-	if err != nil || container.State != "running" {
-		err = CheckRouterPorts()
-		if err != nil {
-			return fmt.Errorf("Unable to listen on required ports, %v,\nTroubleshooting suggestions at https://ddev.readthedocs.io/en/latest/users/troubleshooting/#unable-listen", err)
-		}
+	// Since the ports in use may have changed, stop the router and make sure
+	// we can access all ports.
+	// It might be possible to do this instead by reading from docker all the
+	// existing mapped ports.
+	_, _, err = dockerutil.ComposeCmd([]string{routerComposePath}, "-p", RouterProjectName, "down", "-v")
```

Comment:
Do we actually need to stop the router for this? Shouldn't changes to the ports be picked up by running "docker-compose up" again?

Reply:
docker-compose up would fail if the ports weren't available right? And show some complaints?